### ISSUE TITLE
Could not open JDBC Connection for transaction

### ISSUE THREAD

**Issue by Classical1956** [ClickHouse/clickhouse-jdbc#708]
When using `0.3.1-patch`, I found that it will report an error after being decorated by `@Transactional`.
I know that clickhouse itself does not support transactions, just report this problem.

```
org.springframework.transaction.CannotCreateTransactionException: Could not open JDBC Connection for transaction; nested exception is java.sql.SQLFeatureNotSupportedException: Transactions are not supported
```

**Comment by zhicwu** [comment]
This has been fixed on develop branch.

Now with `jdbc_compliant` option(defaults to true), the driver will fake transaction support. Besides commit() and rollback(), Savepoint can be used as well.

**Comment by zhicwu** [comment]
It's available since 0.3.2-test1.
